src/transformStateWithClones.js

```diff
@@ -3,11 +3,36 @@
 /**
  * @param {Object} state
  * @param {Object[]} actions
- *
- * @return {Object[]}
+ * @returns {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        stateClones.push({ ...stateCopy });
```

Comment:
this action you are doing in all cases, you could move it after switch case

Reply:
sure, my bad. Fixing it now, thanks a lot